nntrainer/layers/layer_node.cpp

```diff
@@ -421,12 +421,9 @@ void LayerNode::read(std::ifstream &file, bool opt_var) {
   if (opt_var) {
     for (unsigned int i = 0; i < run_context->getNumWeights(); ++i) {
       if (run_context->isGradientLastAccess(i) && getTrainable()) {
-        // @note read optimizer variables
-        if (run_context->weightHasGradient(i)) {
-          for (unsigned int j = 0; j < run_context->getNumWeightOptVar(i);
-               ++j) {
-            run_context->getWeightOptVar(i, j).read(file);
-          }
+        /// @note read optimizer variables
+        for (unsigned int j = 0; j < run_context->getNumWeightOptVar(i); ++j) {
+          run_context->getWeightOptVar(i, j).read(file);
```

Comment:
I wonder whether this is working for the bn layer without weightHasGradient check. Cause I had error last time.

Reply:
bn mu and var is non-trainable so it should not allocate any weiht opt var (making run_context->getNumWeightOptVar(i) == 0

https://github.com/nnstreamer/nntrainer/blob/main/nntrainer/layers/bn_layer.cpp#L105-L116

When I tested there was no problem regarding this matter, but if there is we will need to look at :)